chartfx-chart/src/main/java/de/gsi/chart/renderer/spi/ContourDataSetCache.java

```diff
@@ -0,0 +1,460 @@
+package de.gsi.chart.renderer.spi;
+
+import static de.gsi.dataset.DataSet.DIM_X;
+import static de.gsi.dataset.DataSet.DIM_Y;
+import static de.gsi.dataset.DataSet.DIM_Z;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+import javafx.scene.image.Image;
+import javafx.scene.image.PixelFormat;
+import javafx.scene.image.PixelWriter;
+import javafx.scene.image.WritableImage;
+
+import de.gsi.chart.XYChart;
+import de.gsi.chart.axes.Axis;
+import de.gsi.chart.axes.AxisTransform;
+import de.gsi.chart.renderer.ContourType;
+import de.gsi.chart.renderer.datareduction.DefaultDataReducer3D;
+import de.gsi.chart.renderer.datareduction.ReductionType;
+import de.gsi.chart.renderer.spi.utils.ColorGradient;
+import de.gsi.dataset.DataSet;
+import de.gsi.dataset.DataSet3D;
+import de.gsi.dataset.spi.DataRange;
+import de.gsi.dataset.utils.ArrayCache;
+import de.gsi.dataset.utils.CachedDaemonThreadFactory;
+import de.gsi.dataset.utils.ProcessingProfiler;
+
+/**
+ * @author rstein
+ */
+class ContourDataSetCache {
+    private static final String PARALLEL_WORKER_ERROR = "one parallel worker thread finished execution with error";
+    private static final String CLASS_NAME = ContourDataSetCache.class.getSimpleName() + System.currentTimeMillis();
+    private static final String DATA_COPY_BUFFER_NAME = CLASS_NAME + ":dataBuffer";
+    private static final String TEMP_DATA_COPY_BUFFER_NAME = CLASS_NAME + ":tempDataBuffer";
+    private static final String BGRA_BYTE_BUFFER_NAME = CLASS_NAME + ":bgraByteBuffer";
+    private static final int BGRA_BYTE_SIZE = 4;
+    private static final int REF_WIDTH_PARALLEL = 1024;
+    private static final int REF_HEIGHT_PARALLEL = 1000;
+
+    protected final DataSet dataSet;
+    protected final Axis xAxis;
+    protected final Axis yAxis;
+    protected final Axis zAxis;
+    private final ContourDataSetRenderer renderer;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'renderer' could be replaced by a local variable.](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=4880492)